### PR TITLE
Exclude pii from spark enrich and formatting

### PR DESCRIPTION
**Pull request by knservis**

### CODE REVIEW

**Comment by knservis** [comment]
@BenFradet The actual code (not test) change for this ticket is one line: https://github.com/snowplow/snowplow/pull/3633/commits/1c955e12c8b0740f0ca251f02406b74023be2790#diff-9e5742dff28d1d166dcb990bcdaa5a79R210

**Comment by BenFradet** [comment]
then lgtm

when asking for review and the pr is noisy, please pinpoint what you want reviewed or make it less noisy, thanks

**Comment by knservis** [comment]
@BenFradet rebased. It should be clearer now

File: 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  sequential
+  "A job which processes a CloudFront file" should{
+      "filter out pii events" in
+    {
+      runEnrichJob(Lines(MasterCfSpec.lines: _*), "cloudfront", "1", false, List("geo"))
+      val Some(goods) = readPartFile(dirs.output)
+      val Some(bads) = readPartFile(dirs.badRows)
+      goods must not(contain("pii"))
+      bads must not(contain("pii"))
+    }}
```
Comment (BenFradet) [inline 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala]: nit: weird indent / spacing

File: 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+/** Pii filtering test which runs using all of the individual good, bad and misc tests */
+class PiiFilteringSpec extends Specification with EnrichJobSpec {
+  import EnrichJobSpec._
+  override def appName = "pii-filter-cf"
+  sequential
+  "A job which processes a CloudFront file" should{
+      "filter out pii events" in
+    {
+      runEnrichJob(Lines(MasterCfSpec.lines: _*), "cloudfront", "1", false, List("geo"))
```
Comment (BenFradet) [inline 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala]: not the right lines

File: 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      val Some(goods) = readPartFile(dirs.output)
+      val Some(bads) = readPartFile(dirs.badRows)
+      goods must not(contain("pii"))
+      bads must not(contain("pii"))
```
Comment (BenFradet) [inline 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala]: I feel like this is checking the list, not the content of each element?
Also, why would it contain "pii"?

File: 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala (hunk 1 above)
Comment (knservis) [reply 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala]: Should I add scalafmt here too (already doing it in stream enrich)?

Comment (BenFradet) [reply 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala]: sure :+1:

**Comment by knservis** [comment]
@BenFradet @chuwy Is there more to be done here?

File: 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala (hunk 2 above)
Comment (knservis) [reply 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala]: Thanks for that. The point was to use the MasterCfSpec (the lines above were not necessary)

File: 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala (hunk 3 above)
Comment (knservis) [reply 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala]: If I undo the code changes, the test (on good) fails. I thought that means that it matches the content of the field. Let me have another look. About bad, probably the test is redundant currently but there should not be any changes in implementation in the future that cause this to fail. I can remove it..

Comment (knservis) [reply 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/PiiFilteringSpec.scala]: This works but due to a bug in specs2 (https://github.com/etorreborre/specs2/issues/416) Changed

File: 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/EnrichJobSpec.scala [hunk 4]
```diff
@@ -97,7 +90,7 @@ object EnrichJobSpec {
     def apply[S <: String](actual: Expectable[S]) = {
       result((unmatcheable && expected == null) || actual.value == expected,
         "%s (index: %s): %s ".format(field, index, if (unmatcheable) "is unmatcheable" else "%s equals %s".format(actual.description, expected)),
-        "%s (index: %s): %s does not equal %s".format(field, index,  actual.description, expected),
+        "%s (index: %s): %s does not equal %s".format(field, index, actual.description, expected),
```
Comment (chuwy) [inline 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/EnrichJobSpec.scala]: I know it wasn't added in this commit, but can we change `format` to string-interpolation.

**Comment by knservis** [comment]
@BenFradet Is something remaining for your green tick?

File: 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/EnrichJobSpec.scala (hunk 4 above)
Comment (knservis) [reply 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/EnrichJobSpec.scala]: I have no problem changing it. @BenFradet The original (458f2e8cf0) commit uses format instead fo string interpolation. Any reason to prefer format here?

Comment (BenFradet) [reply 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/EnrichJobSpec.scala]: this is just inherited from the scalding code, I don't have any reason

**Comment by BenFradet** [comment]
nope, just forgot, done :+1:

File: 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/EnrichJobSpec.scala (hunk 4 above)
Comment (knservis) [reply 3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/EnrichJobSpec.scala]: Done

**Comment by knservis** [comment]
@chuwy @BenFradet Thanks for the review. Merging

**Pull request merged**